contents/handbook/engineering/clickhouse/clusters.mdx

```diff
@@ -0,0 +1,220 @@
+---
+title: ClickHouse Clusters 
+---
+
+We have three different ClickHouse clusters here at PostHog:
+
+1. Prod-US: Our main production cluster for the US.
+2. Prod-EU: Our main production cluster for the EU.
+3. Dev: Our development cluster.
+
+## Common features
+
+All clusters have these features in common:
+- They are all managed by the ClickHouse team.
+- They are all on version `23.12.5.81`
+- schema (roughly)
+- data retention policy
+- backup policy
+- monitoring
+- alerting
+
+### Backup Policy
+
+We backup all production tables every day. Once a week we take a full backup and then incremental backups for the rest of the week. We keep backups for 8 days.
+
+Backups are managed through [HouseWatch](https://github.com/PostHog/HouseWatch)
+
+We are able to do point in time recovery between daily backups and replay from Kafka topics.
+
+### CH Version
+
+Currently we run `23.12.5.81` 
+
+We run the same version in:
+- Developer environments
+- CI testing environments
+- Production environments
+
+We do this because ClickHouse is notorious for breaking changes between versions. We've seen issues with query syntax compatibility, data result consistency, and other unexpected issues between upgrades.
+
+In order to upgrade ClickHouse, we need to bump the version on CI to test for regressions and compatibility issues. We do this by **adding** the ClickHouse version to the CI Matrix of ClickHouse versions.
+This has the issue of slowing down CI because we then run two tests for everything on the current and desired version on ClickHouse, but it works nicely because it shows the discrepancies between the versions clearly.
+
+Once we have resolved all issues on CI, we can then upgrade the ClickHouse version on:
+- Developer environments
+- Prod US
+- Prod EU
+- Dev cluster
+
+## Prod-US
+
+Prod US is our main production cluster for the US.
+
+It is made up of the following topology:
+
+- 2 shards
+- 3 replicas
+
+### Online Cluster
+
+2 out of the 3 replicas are what we call the 'Online cluster'. It serves all traffic coming in from us.posthog.com. We do this to guard against background tasks consuming resources and slowing down query times on the app. We've seen query time variability otherwise.
```

Comment:
Could you provide a link to take a look at how the load balancers are configured to cover this? I guess we would have two load balancers that target a different set of instances, wouldn't we?